plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/functions/tablechanges/TableChangesFunction.java

```diff
@@ -0,0 +1,132 @@
+import io.trino.spi.connector.ConnectorTableHandle;
+import io.trino.spi.connector.ConnectorTransactionHandle;
+import io.trino.spi.connector.SchemaTableName;
+import io.trino.spi.connector.TableNotFoundException;
+import io.trino.spi.function.SchemaFunctionName;
+import io.trino.spi.ptf.AbstractConnectorTableFunction;
+import io.trino.spi.ptf.Argument;
+import io.trino.spi.ptf.Descriptor;
+import io.trino.spi.ptf.ScalarArgument;
+import io.trino.spi.ptf.ScalarArgumentSpecification;
+import io.trino.spi.ptf.TableFunctionAnalysis;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static io.trino.plugin.base.util.Functions.checkFunctionArgument;
+import static io.trino.plugin.deltalake.DeltaLakeColumnType.SYNTHESIZED;
+import static io.trino.spi.StandardErrorCode.INVALID_FUNCTION_ARGUMENT;
+import static io.trino.spi.ptf.ReturnTypeSpecification.GenericTable.GENERIC_TABLE;
+import static io.trino.spi.type.BigintType.BIGINT;
+import static io.trino.spi.type.TimestampWithTimeZoneType.TIMESTAMP_TZ_MILLIS;
+import static io.trino.spi.type.VarcharType.VARCHAR;
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+
+public class TableChangesFunction
+        extends AbstractConnectorTableFunction
+{
+    private static final String SCHEMA_NAME = "system";
+    private static final String NAME = "table_changes";
+    public static final SchemaFunctionName TABLE_CHANGES_NAME = new SchemaFunctionName(SCHEMA_NAME, NAME);
+    public static final String SCHEMA_NAME_ARGUMENT = "SCHEMA_NAME";
+    private static final String TABLE_NAME_ARGUMENT = "TABLE_NAME";
+    private static final String SINCE_VERSION_ARGUMENT = "SINCE_VERSION";
+    private static final String CHANGE_TYPE_COLUMN_NAME = "_change_type";
+    private static final String COMMIT_VERSION_COLUMN_NAME = "_commit_version";
+    private static final String COMMIT_TIMESTAMP_COLUMN_NAME = "_commit_timestamp";
+
+    private final DeltaLakeMetadataFactory deltaLakeMetadataFactory;
+
+    public TableChangesFunction(DeltaLakeMetadataFactory deltaLakeMetadataFactory)
+    {
+        super(
+                SCHEMA_NAME,
+                NAME,
+                ImmutableList.of(
+                        ScalarArgumentSpecification.builder().name(SCHEMA_NAME_ARGUMENT).type(VARCHAR).build(),
+                        ScalarArgumentSpecification.builder().name(TABLE_NAME_ARGUMENT).type(VARCHAR).build(),
+                        ScalarArgumentSpecification.builder().name(SINCE_VERSION_ARGUMENT).type(BIGINT).defaultValue(-1L).build()),
+                GENERIC_TABLE);
+        this.deltaLakeMetadataFactory = requireNonNull(deltaLakeMetadataFactory, "deltaLakeMetadataFactory is null");
+    }
+
+    @Override
+    public TableFunctionAnalysis analyze(ConnectorSession session, ConnectorTransactionHandle transaction, Map<String, Argument> arguments)
```

Comment:
@homar where do we make sure that we don't read table changes from a table on which we are not supposed to be reading ? 

I don't seem to find anywhere an `AccessControl` reference.

cc @kokosing @kasiafi

Reply:
as discussed offline, there is access control check but it checks if the function can be called, not specific table.
I added a test for this

Reply:
Sorry for misleading you.
I've read more carefully https://trino.io/docs/current/security/file-system-access-control.html

> It’s a good practice to limit access to query table function because this table function works like a query passthrough and ignores tables rules. The following example allows the admin user to execute query table function from any catalog

I realised that the table functions need to be explicitly activated for a user/group of users to be used.
No action needed in this PR.

Reply:
I think it would be nice to pass `AccessControl` to this `analyze` method  so in case of some functions we could follow the access control rules. Like we do `SyncPartitionMetadataProcedure`. I guess it could be out of the scope of this PR.

Reply:
Created follow-up https://github.com/trinodb/trino/issues/17226